addons/cookoff/functions/fnc_detonateAmmunition.sqf

```diff
@@ -101,9 +102,25 @@ if (_amountOfMagazines > 0) exitWith {
             "ACE_ammoExplosionLarge" createvehicle (_vehicle modelToWorld _effect2pos);
         };
     };
-    if (toLower _simType in ["shotdirectionalbomb", "shotilluminating", "shotmine"]) then {
+    if (toLower _simType in ["shotdirectionalbomb", "shotmine"]) then {
         if (random 1 < 0.5) then {
-            [_vehicle, _ammo, 0, false] call _spawnProjectile;
+            // Not all explosives detonate on destruction, some have scripted alternatives
+            private _scripted = getNumber (_ammoCfg >> "triggerWhenDestroyed") == 1;
+            if !(_scripted) then {
+                _ammo = getText (_ammoCfg >> "ace_explosives_Explosive");
+            };
+
+            // If a scripted alternative doesn't exist use generic explosion
+            if (_ammo != "") then {
+                [_vehicle, _ammo, 0, false] call _spawnProjectile;
+            } else {
+                "SmallSecondary" createvehicle (_vehicle modelToWorld _effect2pos);
```

Comment:
Fair, was just following the existing uses of `createVehicle` in the cook off code. Will change them all while I'm at it.